### ISSUE TITLE
SCRAM-SHA-1 Support

### ISSUE THREAD

**Issue by dustinblackman** [mafintosh/mongojs#208]
Mongo 3 uses SCRAM-SHA-1 for auth by default, and MongoJS currently only auths using MongoCR without an option to use otherwise. http://docs.mongodb.org/manual/core/authentication/#authentication-mechanisms

https://github.com/mafintosh/mongojs/blob/master/lib/get-topology.js#L32

Waiting on: 
https://github.com/christkv/mongodb-core/pull/28

**Comment by kzap** [comment]
+1

tested and using this, it works, pls merge. mongo 3.0 cant use mongocr once it starts using scram-sha-1

**Comment by colinbes** [comment]
I see mention of passing option for authMechanism

```
mongojs('username:password@localhost/bcdb', ['mycollection'], {authMechanism: 'ScramSHA1'});
```

but how does one specify the authority source database?

In mongodb it's defined in the url along with the authentication mechanism.

```
"mongodb://username:password@localhost:27017/bcdb/?authSource=bcdb&authMechanism=SCRAM-SHA-1"
```

Thanks
Colin

**Comment by colinbes** [comment]
Do I need to open a new ticket for this (I thought it better to keep it on this ticket as it's relevant) as I don't see how to specify the authentication source database which makes mongojs unusable in many situations - have I missed something?

**Comment by saintedlama** [comment]
@colinbes could you open a new

**Comment by saintedlama** [comment]
@colinbes could you open a new issue for the authority database?
